### PR TITLE
(fix) O3-3271: Fix Question Position Changes on Edit in Form Builder

### PR DESCRIPTION
**Pull request by NethmiRodrigo**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR fixes an issue with the logic that handles editing questions in the Interactive Builder. The underlying cause was the same as of issue O3-3272, which was fixed in PR #280, which because the `pageIndex`, `sectionIndex` and `questionIndex` parameters of the deleteQuestion function are not being correctly set.

The solution was the same as of the mentioned PR, which was to register the EditQuestion dialog as a [modal](https://github.com/openmrs/openmrs-esm-core/pull/961).

## Screenshots

https://github.com/openmrs/openmrs-esm-form-builder/assets/34070216/df1c25fe-8bf3-4206-97d4-f17da895f432

## Related Issue

[O3-3271](https://issues.openmrs.org/browse/O3-3271)

## Other

[O3-3271]: https://openmrs.atlassian.net/browse/O3-3271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/interactive-builder/draggable-question.component.tsx [hunk 1]
```diff
@@ -90,11 +101,7 @@ const DraggableQuestion: React.FC<DraggableQuestionProps> = ({
           hasIconOnly
           iconDescription={t('editQuestion', 'Edit question')}
           kind="ghost"
-          onClick={() => {
-            if (!isDragging) {
-              handleEditButtonClick(question);
-            }
```
Comment (denniskigen) [inline src/components/interactive-builder/draggable-question.component.tsx:96]: Should we factor in `isDragging` when determining whether to launch the modal?

Comment (NethmiRodrigo) [reply src/components/interactive-builder/draggable-question.component.tsx:96]: Would there be a scenario where the edit button gets clicked while a question is being dragged?

Comment (denniskigen) [reply src/components/interactive-builder/draggable-question.component.tsx:96]: Yeah, that makes sense.

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @NethmiRodrigo.

**Pull request merged**